### ISSUE TITLE
Building with VS2022 fails.

### ISSUE THREAD

**Issue by ifratric** [intelxed/xed#286]
When attempting to build from `x64 Native Tools Command Prompt for VS 2022`, the build fails. Full output below.

It appears that mbuild fails to locate `cl.exe` (and later in the process, also `lib.exe`) correctly. This can be fixed by manually specifying paths in the --cc, --cxx, --linker and --ar flags, but this obviously isn't very convenient, especially if xed is used as a part of the larger project.

Additionally, when trying to build from a normal command prompt (outside of the VS command prompt), mbuild fails to locate Visual Studio. I realize building from VS command prompt is recommended, but this used to work for older VS versions.

```
D:\work\xedtest\mbuild>python ..\xed\mfile.py
[MBUILD WARNING] Could not find 64b RC command in SDK directory; assuming on PATH
[PYTHON VERSION] 3.8.10
[XED VERSION] v2022.04.17
[EXTF PROCESSING] ../xed/datafiles/files.cfg
[EXTF PROCESSING] ../xed/datafiles/files-xregs.cfg
[EXTF PROCESSING] ../xed/datafiles/via/files-via-padlock.cfg
[EXTF PROCESSING] ../xed/datafiles/amd/files-amd.cfg
[EXTF PROCESSING] ../xed/datafiles/amd/amdxop/files.cfg
[EXTF PROCESSING] ../xed/datafiles/mpx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/cet/files.cfg
[EXTF PROCESSING] ../xed/datafiles/rdrand/files.cfg
[EXTF PROCESSING] ../xed/datafiles/glm/files.cfg
[EXTF PROCESSING] ../xed/datafiles/sha/files.cfg
[EXTF PROCESSING] ../xed/datafiles/xsaveopt/files.cfg
[EXTF PROCESSING] ../xed/datafiles/xsaves/files.cfg
[EXTF PROCESSING] ../xed/datafiles/xsavec/files.cfg
[EXTF PROCESSING] ../xed/datafiles/clflushopt/files.cfg
[EXTF PROCESSING] ../xed/datafiles/rdseed/files.cfg
[EXTF PROCESSING] ../xed/datafiles/fsgsbase/files.cfg
[EXTF PROCESSING] ../xed/datafiles/smap/files.cfg
[EXTF PROCESSING] ../xed/datafiles/sgx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/rdpid/files.cfg
[EXTF PROCESSING] ../xed/datafiles/pt/files.cfg
[EXTF PROCESSING] ../xed/datafiles/tremont/files.cfg
[EXTF PROCESSING] ../xed/datafiles/movdir/files.cfg
[EXTF PROCESSING] ../xed/datafiles/waitpkg/files.cfg
[EXTF PROCESSING] ../xed/datafiles/cldemote/files.cfg
[EXTF PROCESSING] ../xed/datafiles/sgx-enclv/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx/files.cfg
[Clearing file list for type dec-spine: [ ../xed/datafiles/xed-spine.txt ]]
[EXTF PROCESSING] ../xed/datafiles/ivbavx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/hswavx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/hswbmi/files.cfg
[EXTF PROCESSING] ../xed/datafiles/hsw/files.cfg
[EXTF PROCESSING] ../xed/datafiles/bdw/files.cfg
[EXTF PROCESSING] ../xed/datafiles/skl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/skx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/pku/files.cfg
[EXTF PROCESSING] ../xed/datafiles/clwb/files.cfg
[EXTF PROCESSING] ../xed/datafiles/clx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/vnni/files.cfg
[EXTF PROCESSING] ../xed/datafiles/cpx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512-bf16/files.cfg
[EXTF PROCESSING] ../xed/datafiles/knl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/knm/files.cfg
[EXTF PROCESSING] ../xed/datafiles/4fmaps-512/files.cfg
[EXTF PROCESSING] ../xed/datafiles/4vnniw-512/files.cfg
[EXTF PROCESSING] ../xed/datafiles/vpopcntdq-512/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512f/shared-files.cfg
[Clearing file list for type dec-spine: [ ../xed/datafiles/avx/avx-spine.txt ]]
[CONSIDERING SOURCE] D:\work\xedtest\xed\datafiles\knc\xed-operand-values-interface-uisa.c source 1
[ADDING SOURCE] D:\work\xedtest\xed\datafiles\knc\xed-operand-values-interface-uisa.c source 1
[EXTF PROCESSING] ../xed/datafiles/avx512f/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512cd/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512-skx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/cnl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512ifma/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512vbmi/files.cfg
[EXTF PROCESSING] ../xed/datafiles/icl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/wbnoinvd/files.cfg
[EXTF PROCESSING] ../xed/datafiles/pconfig/files.cfg
[EXTF PROCESSING] ../xed/datafiles/bitalg/files.cfg
[EXTF PROCESSING] ../xed/datafiles/vbmi2/files.cfg
[EXTF PROCESSING] ../xed/datafiles/gfni-vaes-vpcl/files-sse.cfg
[EXTF PROCESSING] ../xed/datafiles/gfni-vaes-vpcl/files-avx-avx512.cfg
[EXTF PROCESSING] ../xed/datafiles/vpopcntdq-vl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/tgl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/vp2intersect/files.cfg
[EXTF PROCESSING] ../xed/datafiles/keylocker/files.cfg
[EXTF PROCESSING] ../xed/datafiles/adl/files.cfg
[EXTF PROCESSING] ../xed/datafiles/hreset/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx-vnni/files.cfg
[EXTF PROCESSING] ../xed/datafiles/spr/files.cfg
[EXTF PROCESSING] ../xed/datafiles/uintr/files.cfg
[EXTF PROCESSING] ../xed/datafiles/amx-spr/files.cfg
[EXTF PROCESSING] ../xed/datafiles/enqcmd/files.cfg
[EXTF PROCESSING] ../xed/datafiles/tsx-ldtrk/files.cfg
[EXTF PROCESSING] ../xed/datafiles/serialize/files.cfg
[EXTF PROCESSING] ../xed/datafiles/tdx/files.cfg
[EXTF PROCESSING] ../xed/datafiles/avx512-fp16/files.cfg
[EXTF PROCESSING] ../xed/datafiles/evex-map5-6/files.cfg
[EXTF PROCESSING] ../xed/datafiles/future/files.cfg
[EMIT BUILD DEFINES HEADER FILE]
R: 0 P: 0 C: 0 E: 0 / 55 msecs
R: 4 P: 109 C: 0 E: 0 / 33 msecs
[MBUILD WARNING] Command execution failed. Waiting for remaining jobs and exiting.
R: 3 P: 109 C: 1 E: 1 / 48 msecs
[COMMAND     ] "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/bin/amd64/cl.exe" -I../xed/include/private -I../xed/include/public/xed -I../xed/include/public -Iobj -Iobj/include-private     /nologo  /MT  /favor:EM64T  /W4 /WX /wd4091 /wd4127 /wd4505 /wd4702 /wd4244 /wd4292 /DXED_VERSION="v2022.04.17" /DXED_AMD_ENABLED /DXED_VIA_ENABLED /DXED_AVX /DXED_SUPPORTS_AVX512 /DXED_MPX /DXED_CET /DXED_SUPPORTS_SHA /DXED_SUPPORTS_WBNOINVD /DXED_DECODER /DXED_ENCODER /DXED_SUPPORTS_LZCNT_TZCNT /DXED_BUILD /c /Foobj/xed-attribute-enum.obj D:/work/xedtest/mbuild/obj/xed-attribute-enum.c
[EXIT_STATUS ] 399
[STDERR]

COMMAND ENCOUNTERD AN EXCEPTION
Traceback (most recent call last):
  File "D:\work\xedtest\mbuild\..\xed\..\mbuild\mbuild\util.py", line 1039, in run
    self.sub = subprocess.Popen(cmd_args,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

[MBUILD WARNING] Command execution failed. Waiting for remaining jobs and exiting.
R: 2 P: 109 C: 2 E: 2 / 50 msecs
[COMMAND     ] "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/bin/amd64/cl.exe" -I../xed/include/private -I../xed/include/public/xed -I../xed/include/public -Iobj -Iobj/include-private     /nologo  /MT  /favor:EM64T  /W4 /WX /wd4091 /wd4127 /wd4505 /wd4702 /wd4244 /wd4292 /DXED_VERSION="v2022.04.17" /DXED_AMD_ENABLED /DXED_VIA_ENABLED /DXED_AVX /DXED_SUPPORTS_AVX512 /DXED_MPX /DXED_CET /DXED_SUPPORTS_SHA /DXED_SUPPORTS_WBNOINVD /DXED_DECODER /DXED_ENCODER /DXED_SUPPORTS_LZCNT_TZCNT /DXED_BUILD /c /Foobj/xed-address-width-enum.obj D:/work/xedtest/mbuild/obj/xed-address-width-enum.c
[EXIT_STATUS ] 399
[STDERR]

COMMAND ENCOUNTERD AN EXCEPTION
Traceback (most recent call last):
  File "D:\work\xedtest\mbuild\..\xed\..\mbuild\mbuild\util.py", line 1039, in run
    self.sub = subprocess.Popen(cmd_args,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

[MBUILD WARNING] Command execution failed. Waiting for remaining jobs and exiting.
R: 1 P: 109 C: 3 E: 3 / 51 msecs
[COMMAND     ] "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/bin/amd64/cl.exe" -I../xed/include/private -I../xed/include/public/xed -I../xed/include/public -Iobj -Iobj/include-private     /nologo  /MT  /favor:EM64T  /W4 /WX /wd4091 /wd4127 /wd4505 /wd4702 /wd4244 /wd4292 /DXED_VERSION="v2022.04.17" /DXED_AMD_ENABLED /DXED_VIA_ENABLED /DXED_AVX /DXED_SUPPORTS_AVX512 /DXED_MPX /DXED_CET /DXED_SUPPORTS_SHA /DXED_SUPPORTS_WBNOINVD /DXED_DECODER /DXED_ENCODER /DXED_SUPPORTS_LZCNT_TZCNT /DXED_BUILD /c /Foobj/xed-reg-enum.obj D:/work/xedtest/mbuild/obj/xed-reg-enum.c
[EXIT_STATUS ] 399
[STDERR]

COMMAND ENCOUNTERD AN EXCEPTION
Traceback (most recent call last):
  File "D:\work\xedtest\mbuild\..\xed\..\mbuild\mbuild\util.py", line 1039, in run
    self.sub = subprocess.Popen(cmd_args,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

[MBUILD WARNING] Command execution failed. Waiting for remaining jobs and exiting.
R: 0 P: 109 C: 4 E: 4 / 53 msecs
[COMMAND     ] "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/bin/amd64/cl.exe" -I../xed/include/private -I../xed/include/public/xed -I../xed/include/public -Iobj -Iobj/include-private     /nologo  /MT  /favor:EM64T  /W4 /WX /wd4091 /wd4127 /wd4505 /wd4702 /wd4244 /wd4292 /DXED_VERSION="v2022.04.17" /DXED_AMD_ENABLED /DXED_VIA_ENABLED /DXED_AVX /DXED_SUPPORTS_AVX512 /DXED_MPX /DXED_CET /DXED_SUPPORTS_SHA /DXED_SUPPORTS_WBNOINVD /DXED_DECODER /DXED_ENCODER /DXED_SUPPORTS_LZCNT_TZCNT /DXED_BUILD /c /Foobj/xed-operand-ctype-enum.obj D:/work/xedtest/mbuild/obj/xed-operand-ctype-enum.c
[EXIT_STATUS ] 399
[STDERR]

COMMAND ENCOUNTERD AN EXCEPTION
Traceback (most recent call last):
  File "D:\work\xedtest\mbuild\..\xed\..\mbuild\mbuild\util.py", line 1039, in run
    self.sub = subprocess.Popen(cmd_args,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

ABORT: Library build failed
```

**Comment by expend20** [comment]
hey @ifratric, just use this pull https://github.com/intelxed/mbuild/pull/35

**Comment by marjevan** [comment]
Hi,
We'll release a new mbuild version with official VS2022 support soon.

Thanks.

**Comment by marjevan** [comment]
mbuild with VS2022 support is available on version v2022.04.17 https://github.com/intelxed/mbuild/commit/b41485956bf65d51b8c2379768de7eaaa7a4245b.
